lib/atomic_cache/storage/dalli.rb

```diff
@@ -33,12 +33,14 @@ def add(key, new_value, ttl, user_options=nil)
 
       def read(key, user_options=nil)
```

Comment:
should user_options default to `{}` instead of nil?

Alternatively, do keyword args make more sense?

Reply:
keyword args would add a level of explicitness, which I think would be a good improvement

Reply:
Ah, actually, I remember now why I didn't do that originally.  It's because these options end up getting passed into the dalli client.  If there was a way to say "I expect these options, but also take a hash of unknown options that I'll just pass on" that'd be cool, but I wasn't aware of such a language mechanic.

Reply:
Just to note, use of kwargs would drop support for Ruby < 2.0. But 1.9.3 has been EOL'ed so I don't think we will need to worry about it.

Reply:
Would `raw` impact any of the downstream clients? I wonder if we want to explicitly separate those like `cache_client_options` and then `raw` kwarg for things we understand?

Reply:
Regarding `=nil` vs `={}`, the one challenge I see to doing `={}` is if `nil` gets passed in, you don't automatically get a `{}` out of it:

```
irb(main):001:0> def foo(arg={})
irb(main):002:1> puts arg
irb(main):003:1> end
=> :foo
irb(main):004:0> foo(nil)

=> nil
```

vs.

```
irb(main):005:0> def foo2(arg=nil)
irb(main):006:1> arg ||= {}
irb(main):007:1> puts arg
irb(main):008:1> end
=> :foo2
irb(main):009:0> foo2(nil)
{}
```

I think one argument in favor of `={}` is that if the user is explicitly passing in `nil`, then maybe an exception is the right response there, as otherwise the latter form may be hiding a bug.  Thoughts?

Reply:
I definitely see your point that this requires an additional guard clause to prevent errors from bubbling up. However, I tend to agree that if the documentation says you should pass something `Hash`-y and you pass `nil`, you're in violation of the API. I don't feel strongly on the `nil` || `{}` issue, so definitely feel free to go with what feels best to you :-)